velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -719,6 +894,14 @@ static bool compareMaterializedRows(
     }
   }
 
+  // In case there are values in right but not in left, check the other way
+  // around. E.g., left = {1, 1, 2}, right = {1, 2, 3}.
+  for (auto& it : right) {
```

Comment:
This is a nice catch, but seems separate from the other issue you are fixing. It would be nice to extract it into a separate PR.

Reply:
Yeah, this is fixed in https://github.com/facebookincubator/velox/pull/3725.